sdk/bcs/src/bcs-type.ts

```diff
@@ -69,14 +72,44 @@ export class BcsType<T, Input = T> {
 		return new BcsType<T2, Input2>({
 			name: name ?? this.name,
 			read: (reader) => output(this.read(reader)),
-			write: (value, writer) => this._write(input(value), writer),
+			write: (value, writer) => this.#write(input(value), writer),
 			serializedSize: (value) => this.serializedSize(input(value)),
-			serialize: (value, options) => this._serialize(input(value), options),
+			serialize: (value, options) => this.#serialize(input(value), options),
 			validate: (value) => this.validate(input(value)),
 		});
 	}
 }
 
+export class SerializedBcs<T, Input = T> {
+	#schema: BcsType<T, Input>;
+	#bytes: Uint8Array;
+
+	constructor(type: BcsType<T, Input>, schema: Uint8Array) {
```

Comment:
Should the type be optional so that you can still construct serialized BCS without the underlying schema? IDK why you'd do that though, or how that would even remotely be useful...

Reply:
can always make it optional in the future 🤷